src/main/java/com/ldtteam/structurize/util/IndexedTranslatableContents.java

```diff
@@ -0,0 +1,155 @@
+package com.ldtteam.structurize.util;
+
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.locale.Language;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.ComponentContents;
+import net.minecraft.network.chat.ComponentUtils;
+import net.minecraft.network.chat.FormattedText;
+import net.minecraft.network.chat.FormattedText.ContentConsumer;
+import net.minecraft.network.chat.FormattedText.StyledContentConsumer;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.contents.TranslatableContents;
+import net.minecraft.world.entity.Entity;
+import javax.annotation.Nullable;
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Inspired by {@link TranslatableContents} to support things from {@link MessageFormat}
+ */
+public class IndexedTranslatableContents implements ComponentContents
+{
```

Comment:
What is this for?

Reply:
Look en_us.json for default keys, it's the classic {index, format, etc} translating instead of dumb %letter thing

Reply:
And we need all of that just to pick up a translation ?

Reply:
Yeah, but it's ten times better cuz % thing is far from decent for more than half of languages

Reply:
Yeah, but I think its not worth to stray from the standard that we are using everywhere

Reply:
we definitely should ditch percent if we can, message format from icu introduces proper plural etc. see <https://support.crowdin.com/icu-message-syntax/>